tests/units/data/test_torch_dataset.py

```diff
@@ -0,0 +1,59 @@
+from typing import Optional
+
+import torch
+
+from docarray import BaseDocument, DocumentArray
+from docarray.data import MultiModalDataset
+from docarray.documents import Image, Text
+
+
+class ImagePreprocess:
+    def __call__(self, image: Image) -> None:
+        assert isinstance(image, Image)
+        image.tensor = torch.randn(3, 32, 32)
+
+
+class TextPreprocess:
+    def __call__(self, text: Text) -> None:
+        assert isinstance(text, Text)
+        text.embedding = torch.randn(64)
+
+
+def test_torch_dataset():
+    BATCH_SIZE = 32
+
+    class PairTextImage(BaseDocument):
+        text: Text
+        image: Image
+
+    class Mydataset(MultiModalDataset[PairTextImage]):
+        def __init__(self, csv_file: str, preprocessing: Optional[dict] = None):
+            with open(csv_file, "r") as f:
+                f.readline()
+                da = DocumentArray[PairTextImage](
+                    PairTextImage(
+                        text=Text(text=i[1]),
+                        image=Image(url=f"tests/toydata/image-data/{i[0]}"),
+                    )
+                    for i in map(lambda x: x.strip().split(","), f.readlines())
+                )
+            super().__init__(da, preprocessing)
```

Comment:
I am not sure to like this pattern so much? Shouldnt' maybe we ask them to implement a `da` property or something like this?

Reply:
Yea, the rest of the team didn't like it either. We decided to go with directly using the `MultiModalDataset` instead. So they have a `da` and a preprocessing dictionary and we give them a dataset.

```python
da = ...
preprocessing = ...
ds = MultiModalDataset[PairTextImage](da, preprocessing)
dl = DataLoader(ds, ...)
...
```

Reply:
If they want something more custom, they can do so. They can still use our collate function.

```python
ds = MyCustomDataset(*args)
loader = DataLoader(
    ds,
    collate_fn=MultiModalDataset[PairTextImage].collate_fn,
    ...
)
```